prow/genjobs/cmd/genjobs/main.go

```diff
@@ -194,6 +203,106 @@ func convertOrgRepoStr(o options, s string) string {
 	return strings.Join([]string{o.orgMap[org], repo}, "/")
 }
 
+// combinePresets reads a list of paths and aggregates the presets.
+func combinePresets(paths []string) []config.Preset {
+	presets := []config.Preset{}
+
+	if len(paths) == 0 {
+		return presets
+	}
+
+	for _, p := range paths {
+		c, err := config.ReadJobConfig(p)
+		if err != nil {
+			continue
+		}
+		presets = append(presets, c.Presets...)
+	}
+
+	return presets
+}
+
+// mergePreset merges a preset into a job Spec based on defined labels.
+func mergePreset(labels map[string]string, job *config.JobBase, preset config.Preset) {
+	for l, v := range preset.Labels {
+		if v2, exists := labels[l]; !exists || v != v2 {
+			return
+		}
+	}
+
+	for _, env := range preset.Env {
+	econtainer:
+		for i := range job.Spec.Containers {
+			for j := range job.Spec.Containers[i].Env {
+				if job.Spec.Containers[i].Env[j].Name == env.Name {
+					job.Spec.Containers[i].Env[j].Value = env.Value
+					continue econtainer
+				}
+			}
+
+			job.Spec.Containers[i].Env = append(job.Spec.Containers[i].Env, env)
+		}
+	}
+
+volume:
+	for _, vol := range preset.Volumes {
+
+		for i := range job.Spec.Volumes {
+			if job.Spec.Volumes[i].Name == vol.Name {
+				job.Spec.Volumes[i] = vol
+				continue volume
+			}
+
+		}
+
+		job.Spec.Volumes = append(job.Spec.Volumes, vol)
+	}
+
+	for _, volm := range preset.VolumeMounts {
+	vcontainer:
+		for i := range job.Spec.Containers {
+			for j := range job.Spec.Containers[i].VolumeMounts {
+				if job.Spec.Containers[i].VolumeMounts[j].Name == volm.Name {
+					job.Spec.Containers[i].VolumeMounts[j] = volm
+					continue vcontainer
+				}
+			}
+
+			job.Spec.Containers[i].VolumeMounts = append(job.Spec.Containers[i].VolumeMounts, volm)
+		}
+	}
+}
+
+// resolvePresets resolves all preset for a particular job Spec based on defined labels.
+func resolvePresets(o options, labels map[string]string, job *config.JobBase, presets []config.Preset) {
```

Comment:
Can we reuse this? https://github.com/kubernetes/test-infra/blob/1e74b899028a9f18076f30138846175fd3112600/prow/config/config.go#L1565-L1575

Reply:
Originally, I was going to create a PR to make that upstream [`resolvePresets(..)`](https://github.com/kubernetes/test-infra/blob/1e74b899028a9f18076f30138846175fd3112600/prow/config/config.go#L1565-L1575) method *public*, and use it. 

Upon starting to make that change, I realized that the version of [`mergePreset(..)`](https://github.com/kubernetes/test-infra/blob/1e74b899028a9f18076f30138846175fd3112600/prow/config/jobs.go#L44:6) that is called does not support *overriding* `env`, `volumes`, of `volumeMounts` values, but instead the signature returns an `error` to handle conflict resolution.

So instead of modifying upstream logic for a specific downstream use-case, I decided to adapt the code.